tests/e2e/lib/diskutils

```diff
@@ -0,0 +1,105 @@
+# along with this program; If not, see <http://www.gnu.org/licenses/>.
+
+select_disk_to_partition(){
+  ###########################################################################
+  # Description:                                                            #
+  # Select free partition with size in "G"                                  #
+  #                                                                         #
+  # Arguments: should be be exported by calling script                      #
+  #   DEVICE,  set free /dev/ device                                        #
+  #   PART_ID, set device partition                                         #
+  ###########################################################################
+
+  local disk_table
+  disk_table=$(lsblk --noheadings --raw)
+  local disks_arr
+  disks_arr=$(echo "$disk_table" | awk '$1~// && $6=="disk" {print $1}')
+  info_message "select_disk_to_partition, found ${disks_arr}"
+  info_message "=============================="
+
+  for disk in $disks_arr; do
+     if [[ ${disk} == "vda" || \
+           $(echo "${disk_table}" | grep -c "${disk}" ) -eq 1 && ${disk} != "zram0" ]];then
+         new_part="$( (echo n; echo p; echo ; echo ; echo ; echo w) | fdisk "/dev/${disk}")"
+         part_id="$(echo "${new_part}" | grep -Po "new partition \K([0-9])")"
+         DISK="${disk}"
+         if [[ $(echo "${disk}" | grep -c nvme) -eq 1 ]]; then
+           part_id=p${part_id}
+           PART_ID="${part_id}"
+           info_message "select_disk_to_partition, found /dev/${DISK}${PART_ID}"
+           info_message "=============================="
+           break;
+         fi
+      fi
+      if [ -e /sys/devices/soc0/machine ]; then
+        if grep -qi SA8775P /sys/devices/soc0/machine; then
+           if [[ ${disk} == "sde" ]]; then
+             DISK=${disk}
+             part=$(echo "${disk_table}" | grep "G" | grep "sde[0-9]." | grep -v "/" | cut -d" " -f1 )
+             PART_ID=${part:3}
+             break;
+           fi
+        fi
+      fi
+  done
+}
+
+create_qm_var_part() {
+  ###########################################################################
+  # Description:                                                            #
+  #  Create relevant filesystem                                             #
+  #    support fstype, non ostree, xfs (c9s vm), ext4(soc)                  #
+  # ARGUMENTS: should be be exported by calling script                      #
+  #   DEVICE,  set free /dev/ device                                        #
+  #   PART_ID, set device partition                                         #
+  ###########################################################################
+
+  select_disk_to_partition
+
+  local slash_var
+  slash_var="/var"
+
+  # Check is SoC aboot exist, else regular image
+  # VM use case
+  if [ -e /sys/devices/soc0/machine ]; then
+        if grep -qi SA8775P /sys/devices/soc0/machine; then
```

Comment:
SA8775P seems a magic name, can we have a constant to describe it?

Reply:
OK